src/mongocrypt-ctx.c

```diff
@@ -901,6 +901,18 @@ _mongocrypt_ctx_init (mongocrypt_ctx_t *ctx,
       }
    }
 
+   /* Check that the kms provider set and the datakey kms provider match. */
+   if (ctx->opts.kek.kms_provider) {
+      if (!((ctx->crypt->opts.kms_providers.need_credentials |
+             ctx->crypt->opts.kms_providers.configured_providers) &
+            ctx->opts.kek.kms_provider)) {
+         return _mongocrypt_ctx_fail_w_msg (
+            ctx,
+            "datakey provider does not match the provider set in "
```

Comment:
```suggestion
            "requested datakey provider was not configured"
```

This error may be propagated up through drivers. Users likely will not find references to libmongocrypt functions helpful.